hs_app_netCDF/tests/test_netcdf_file_update.py

```diff
@@ -0,0 +1,139 @@
+from django.core.exceptions import ValidationError
+
+from rest_framework import status
+
+from hs_core import hydroshare
+from hs_core.views import update_key_value_metadata
+from hs_core.testing import MockIRODSTestCaseMixin
+from hs_core.hydroshare import utils
+
+from hs_app_netCDF.views import update_netcdf_file
+
+
+class TestUpdateNetcdfFile(MockIRODSTestCaseMixin, TestCase):
+
+    def setUp(self):
+        super(TestUpdateNetcdfFile, self).setUp()
+        self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        self.username = 'john'
+        self.password = 'jhmypassword'
+        self.john = hydroshare.create_account(
+            'john@gmail.com',
+            username=self.username,
+            first_name='John',
+            last_name='Clarson',
+            superuser=False,
+            password=self.password,
+            groups=[]
+        )
+
+        self.temp_dir = tempfile.mkdtemp()
+        self.netcdf_file_name = 'netcdf_valid.nc'
+        self.netcdf_file = 'hs_app_netCDF/tests/{}'.format(self.netcdf_file_name)
+        target_temp_netcdf_file = os.path.join(self.temp_dir, self.netcdf_file_name)
+        shutil.copy(self.netcdf_file, target_temp_netcdf_file)
+        self.netcdf_file_obj = open(target_temp_netcdf_file, 'r')
+
+        self.genResource = hydroshare.create_resource(resource_type='GenericResource',
+                                                      owner=self.john,
+                                                      title='Test Resource',
+                                                      metadata=[])
+
+        self.factory = RequestFactory()
+
+    def tearDown(self):
+        super(TestUpdateNetcdfFile, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_update_netcdf_file(self):
+        # create a netcdf resource
+        files = [UploadedFile(file=self.netcdf_file_obj, name=self.netcdf_file_name)]
+        _, _, metadata, _ = utils.resource_pre_create_actions(
+            resource_type='NetcdfResource',
+            resource_title='Snow water equivalent estimation at TWDEF site '
+                           'from Oct 2009 to June 2010',
+            page_redirect_url_key=None,
+            files=files,
+            metadata=None,)
+
+        self.resNetcdf = hydroshare.create_resource(
+            'NetcdfResource',
+            self.john,
+            'Snow water equivalent estimation at TWDEF site from Oct 2009 to June 2010',
+            files=files,
+            metadata=metadata
+            )
+
+        utils.resource_post_create_actions(self.resNetcdf, self.john, metadata)
+        res_metadata = self.resNetcdf.metadata
+
+        # update res core metadata for setting flag
+        self.assertFalse(res_metadata.is_dirty)
+        res_metadata.create_element('subject', value='new keyword')
+        res_metadata.refresh_from_db()
+        self.assertTrue(res_metadata.is_dirty)
+
```

Comment:
We need to include more test for other metadata elements create/update that sets the ```metadata.is_dirty``` to True.